app/controllers/api/v1/targets_controller.rb

```diff
@@ -7,7 +7,11 @@ def index
       end
 
       def create
-        @target = current_user.targets.create!(target_params)
+        if current_user.targets.count < Target::PER_USER_LIMIT
```

Comment:
I would move all this logic into a validation in the target model where I check if there are already `Target::PER_USER_LIMIT` created for that user